src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder report = new StringBuilder();
+        LocalDate localDateFrom = parseDateFromString(dateFrom);
+        LocalDate localDateTo = parseDateFromString(dateTo);
+
+        report.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+
+        for (String currentName : names) {
+
+            int salaryAmount = 0;
+            for (String record : data) {
+                if (record == null || report.isEmpty()) {
```

Comment:
This condition looks a bit strange. You're checking if `record` is `null` or `report` is `empty()`. But `report` is a `StringBuilder` that you've already appended some text to (lines 14-17), so it will never be empty at this point. Also, the `data` array shouldn't contain `null` values according to the task description, so you can safely remove this condition.